src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBMetricsProvider.cs

```
@@ -0,0 +1,135 @@
+            { "Request rate is large", string.Empty },
+            { "PartitionKey value must be supplied for this operation.", "We do not support lease containers with partitions at this time. Please create a new lease collection without partitions." },
+            { "The remote name could not be resolved:", string.Empty },
+            { "Owner resource does not exist", string.Empty },
+            { "The specified document collection is invalid", string.Empty }
+        };
+
+        public CosmosDBMetricsProvider(ILogger logger, Container monitoredContainer, Container leaseContainer, string processorName)
+        {
+            _logger = logger;
+            _monitoredContainer = monitoredContainer;
+            _leaseContainer = leaseContainer;
+            _processorName = processorName;
+        }
+
+        public async Task<CosmosDBTriggerMetrics> GetMetricsAsync()
+        {
+            int partitionCount = 0;
+            long remainingWork = 0;
+
+            try
+            {
+                List<ChangeFeedProcessorState> partitionWorkList = new List<ChangeFeedProcessorState>();
+                ChangeFeedEstimator estimator = _monitoredContainer.GetChangeFeedEstimator(_processorName, _leaseContainer);
+                using (FeedIterator<ChangeFeedProcessorState> iterator = estimator.GetCurrentStateIterator())
+                {
+                    while (iterator.HasMoreResults)
+                    {
+                        FeedResponse<ChangeFeedProcessorState> response = await iterator.ReadNextAsync();
+                        partitionWorkList.AddRange(response);
+                    }
+                }
+
+                partitionCount = partitionWorkList.Count;
+                remainingWork = partitionWorkList.Sum(item => item.EstimatedLag);
+            }
+            catch (Exception e) when (e is CosmosException || e is InvalidOperationException)
+            {
+                if (!TryHandleCosmosException(e))
```

Why do we swallow CosmosDb exceptions? In this case we end up with "ScaleIn". I would expect to throw the exception so this metric would not count in voting.

https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Scale/FunctionsScaleManager.cs#L111

https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Scale/FunctionsScaleManager.cs#L177